DependencyInjection/CompilerPasses/ExpressionLanguageCachePass.php

```diff
@@ -18,6 +18,10 @@ class ExpressionLanguageCachePass implements CompilerPassInterface
      */
     public function process(ContainerBuilder $container)
     {
+        // Last line (the reference to cache.app) triggers a "Constructing service "monolog.logger.cache" from a parent definition is not supported at build time."
+        // By bypassing this pass the container builds, but expression language throws a deprecation notice about not passing a cache interface in the constructor
+        return;
```

Comment:
Taking a look to this, ExpressionLanguage class has some special cases when defining the cache (https://github.com/symfony/symfony/blob/v3.4.35/src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php). This is implemented since symfony 3.2 (that is the deprecation you mentioned in the comment). What Gary did in that part was to add `cache.app` that is a service of `Symfony\Component\Cache\Adapter\ArrayAdapter` (see where these changes have been implemented https://github.com/smartboxgroup/integration-framework-bundle/pull/94/files) and this `cache.app` service has a call to `setLogger` to be able to log the exceptions (see file https://github.com/symfony/symfony/blob/v3.4.35/src/Symfony/Component/Cache/Traits/ArrayTrait.php). The default cache in `ExpressionLanguage` is `Symfony\Component\Cache\Adapter\ArrayAdapter` but without logger. Myabe we can try without logger to see if it works.

Reply:
Did you take a look to this comment?

Reply:
A separate ticket will be created to investigate this part. At the moment is having the same behaviour as with Symfony 2.8